### ISSUE TITLE
task 2 sales prediction

### ISSUE THREAD

**Issue by ShanmugaSundaram-oss** [ShanmugaSundaram-oss/task-2#1]
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score
import matplotlib.pyplot as plt

``
# Assuming dataset has features like 'TV', 'Radio', 'Newspaper', and 'Sales'
data = pd.read_csv('advertising.csv')

# 1. Preprocess the data
X = data[['TV', 'Radio', 'Newspaper']]  # Features (advertising spend)
y = data['Sales']  # Target (sales)

# 2. Split the dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# 3. Train Linear Regression model
model = LinearRegression()
model.fit(X_train, y_train)

# 4. Make predictions
y_pred = model.predict(X_test)

# 5. Evaluate the model
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)
print(f"Mean Squared Error: {mse:.2f}")
print(f"R² Score: {r2:.2f}")

# 6. Visualize actual vs predicted sales
plt.scatter(y_test, y_pred, color='blue', label='Predicted vs Actual')
plt.plot([y_test.min(), y_test.max()], [y_test.min(), y_test.max()], 'r--', lw=2, label='Ideal Fit')
plt.xlabel('Actual Sales')
plt.ylabel('Predicted Sales')
plt.title('Sales Prediction: Actual vs Predicted')
plt.legend()
plt.show()

# Optional: Feature importance (coefficients)
feature_importance = pd.DataFrame({'Feature': X.columns, 'Coefficient': model.coef_})
print("\nFeature Importance:")
print(feature_importance)
